### PR TITLE
Add TestRunResultProcessor for JUnit-style XML output

### PR DESCRIPTION
**Pull request by guusdk**
Smack will be configurable with optional TestRunResultProcessor instances, which will operate on the outcome of the executed tests. This commit adds a processor that generates an XML file that mimics the JUnit-style reporting.

When Smack is configured to use the new processor introduced by this commit (`org.igniterealtime.smack.inttest.util.JUnitXmlTestRunResultProcessor`) then a file naed `test-results.xml` will be created in the directory identified by the system property `logDir`.

### CODE REVIEW

**Comment by guusdk** [comment]
For brownie points, I added a step in the workflow of this repo that processes the JUnit-style XML output, and generates UI elements from that (there's an additional 'check' in this PR, and the summary of the build shows a table, as shown below).

I'd love to be able to group the test outcome per XEP, but I'm not sure if that's possible with this particular tool. I've asked the author in https://github.com/mikepenz/action-junit-report/issues/1080

image

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -87,12 +89,21 @@ jobs:
...
         uses: actions/upload-artifact@v4
+        if: always() # always run even if the previous step fails
         with:
           name: XMPP debug logs
           path: logs/*
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v4
+        if: always() # always run even if the previous step fails
+        with:
+          report_paths: '**/test-results.xml'
```
Comment (Fishbowler) [inline .github/workflows/build.yml:107]:
```suggestion
          report_paths: '**/test-results.xml'
          suiteRegex: '*'
```
My guess at a workaround for https://github.com/mikepenz/action-junit-report/issues/1080

File: src/main/resources/specifications.properties [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+XEP-0481=Content Types in Messages
+XEP-0482=Call Invites
+XEP-0483=HTTP Online Meetings
+XEP-0484=Fast Authentication Streamlining Tokens
+XEP-0485=PubSub Server Information
+XEP-0486=MUC Avatars
+XEP-0487=Host Meta 2 - One Method To Rule Them All
+XEP-0488=MUC Token Invite
+XEP-0489=Reporting Account Affiliations
+XEP-0490=Message Displayed Synchronization"specs.put();
```
Comment (Fishbowler) [inline src/main/resources/specifications.properties:513]: Typo? Should this be just
```suggestion
XEP-0490=Message Displayed Synchronization
```
?

File: .github/workflows/build.yml (hunk 1 above)
Comment (guusdk) [reply .github/workflows/build.yml:107]: > Unexpected input(s) 'suiteRegex', valid inputs are ['token', 'github_token', 'report_paths', 'test_files_prefix', 'exclude_sources', 'suite_regex', 'annotate_only', 'check_annotations', 'update_check', 'check_name', 'commit', 'fail_on_failure', 'require_tests', 'require_passed_tests', 'include_passed', 'check_title_template', 'summary', 'check_retries', 'transformers', 'job_summary', 'detailed_summary', 'annotate_notice', 'follow_symlink', 'job_name', 'annotations_limit', 'truncate_stack_traces']

Trying again with `suite_regex`...

Comment (guusdk) [reply .github/workflows/build.yml:107]: Sadly, that didn't make a difference: 
image

**Comment by Fishbowler** [comment]
The regex change to the job has changed the reporting from

> MoodIntegrationTest.testNotificationAfterFilterChange (Normal)	

to 

> MoodIntegrationTest.XEP-0107: User Mood/MoodIntegrationTest.testNotificationAfterFilterChange (Normal)	

Which is more information, but unsure if it's actually nicer.

**Review by Fishbowler — APPROVED** [review]

**Comment by Fishbowler** [comment]
Merged. We'll change the reporting separately, if we're able.

**Pull request merged**